File: source_code/project4_flutter/lib/shared/models/custom_paging.dart
import 'package:json_annotation/json_annotation.dart';

part 'custom_paging.g.dart';

@JsonSerializable()
class CustomPaging {
  int status;
  int currentPage;
  int totalPages;
  int pageSize;
  int totalCount;
  bool hasPrevious;
  bool hasNext;
  String message;
  dynamic data;

  CustomPaging(
      {required this.status,
      required this.message,
      required this.data,
      required this.currentPage,
      required this.totalPages,
      required this.pageSize,
      required this.totalCount,
      required this.hasPrevious,
      required this.hasNext});

  factory CustomPaging.fromJson(Map<String, dynamic> json) =>
      _$CustomPagingFromJson(json);

  Map<String, dynamic> toJson() => _$CustomPagingToJson(this);
}

File: source_code/project4_spring/main/src/main/java/com/service/main/dto/CustomPaging.java
package com.service.main.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@Data
@NoArgsConstructor
public class CustomPaging {
    private int status;

    private String message;

    private int currentPage;

    private int totalPages;

    private int pageSize;

    private long totalCount;

    private boolean hasPrevious;

    public boolean isHasPrevious() {
        return currentPage > 0;
    }

    private boolean hasNext;

    public boolean isHasNext() {
        return currentPage < totalPages;
    }



    private Object data;
}
